### PR TITLE
Fix PDOConnector GeneratedID return type

### PR DESCRIPTION
**Pull request by johannesx75**
After inserting a DataObject into a database the ID of that object should be an int.
PDOConnector uses PDO::lastInsertId to retrieve the ID. PDO::lastInsertId returns a string. This fix converts the return value to int to make it compliant with the DBConnector::getGeneratedID docs.

To reproduce this issue:
* Create a new DataObject
* Write it to the DB.
* Check the type of the ID Field.

### CODE REVIEW

**Comment by johannesx75** [comment]
This affects SS3 all the way up to current dev-master. The fix works for for SS3 as well. I wasn't sure wich version to target.

File: src/ORM/Connect/PDOConnector.php [hunk 1]
```diff
@@ -483,7 +483,7 @@ public function getLastError()
 
     public function getGeneratedID($table)
     {
-        return $this->pdoConnection->lastInsertId();
+        return intval($this->pdoConnection->lastInsertId());
```
Comment (robbieaverill) [inline src/ORM/Connect/PDOConnector.php]: This is a stylistic thing which may be bordering on opinionated, but I'd probably prefer `(int) $this...` casting over using `intval`. I think that's usually how we cast in other parts of core code.

**Comment by ScopeyNZ** [comment]
Just noting that ID's aren't necessarily integers. In PostgreSQL it's entirely possible to create a non-numeric sequence (eg. a number with a prefix) that generates IDs for records.

File: src/ORM/Connect/PDOConnector.php (hunk 1 above)
Comment (johannesx75) [reply src/ORM/Connect/PDOConnector.php]: I created a second PR (#8874) that uses (int) to do the casting. PSR-2 actually states that (int) should be preferred to intval(). What I could not find is a statement wether there should be a space after type cast. So I just went with your sample code.

**Comment by johannesx75** [comment]
@ScopeyNZ I was going to do the casting further up in the chain. Where the ID is actually written to the DataObject after an insert. But then I noticed that the PHPDoc for DBConnector::getGeneratedID states that it returns an integer. So I figured SilverStripe decided to only use integers as IDs in any Database. I decided to make the code adhere to the docs instead of fixing the docs in DBConnector and also change DataObjects write implementation.

**Comment by kinglozzer** [comment]
Closing in favour of https://github.com/silverstripe/silverstripe-framework/pull/8874

**Pull request closed without merging**